src/asn_index.c

```diff
@@ -0,0 +1,256 @@
+/*
+ * Copyright (c) 2016, OARC, Inc.
+ * Copyright (c) 2007, The Measurement Factory, Inc.
+ * Copyright (c) 2007, Internet Systems Consortium, Inc.
+ * All rights reserved.
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions
+ * are met:
+ *
+ * 1. Redistributions of source code must retain the above copyright
+ *    notice, this list of conditions and the following disclaimer.
+ *
+ * 2. Redistributions in binary form must reproduce the above copyright
+ *    notice, this list of conditions and the following disclaimer in
+ *    the documentation and/or other materials provided with the
+ *    distribution.
+ *
+ * 3. Neither the name of the copyright holder nor the names of its
+ *    contributors may be used to endorse or promote products derived
+ *    from this software without specific prior written permission.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+ * "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+ * LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS
+ * FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE
+ * COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT,
+ * INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING,
+ * BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+ * LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+ * CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
+ * LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN
+ * ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ * POSSIBILITY OF SUCH DAMAGE.
+ */
+
+#include "config.h"
+
+#if HAVE_LIBGEOIP
+
+#include <stdlib.h>
+#include <assert.h>
+#include <stdio.h>
+#include <string.h>
+#include <ctype.h>
+#include <GeoIP.h>
+
+#include "xmalloc.h"
+#include "dns_message.h"
+#include "md_array.h"
+#include "hashtbl.h"
+#include "syslog_debug.h"
+
+extern int debug_flag;
+extern char * geoip_asn_v4_dat;
+extern int geoip_asn_v4_options;
+extern char * geoip_asn_v6_dat;
+extern int geoip_asn_v6_options;
+static hashfunc asn_hashfunc;
+static hashkeycmp asn_cmpfunc;
+
+#define MAX_ARRAY_SZ 65536
+static hashtbl *theHash = NULL;
+static int next_idx = 0;
+static GeoIP *geoip = NULL;
+static GeoIP *geoip6 = NULL;
+static char *ipstr;
+static char *nodb = "noGeoIPDB";
+static char *unknown = "unknown";
+static char *unknown_v4 = "unknown IPv4";
+static char *unknown_v6 = "unknown IPv6";
```

Comment:
Can you change the above to be like country indexer and the no db to `NODB`, would like it to be the same and I am very unsure how well whitespace works in the labels and all the way to the Presenter.